src/closure.rs

```diff
@@ -460,10 +454,8 @@ impl Closure<dyn FnOnce()> {
 /// A trait for converting an `FnOnce(A...) -> R` into a `FnMut(A...) -> R` that
 /// will throw if ever called more than once.
 #[doc(hidden)]
-pub trait WasmClosureFnOnce<A, R>: 'static {
-    type FnMut: ?Sized + 'static + WasmClosure;
-
-    fn into_fn_mut(self) -> Box<Self::FnMut>;
+pub trait WasmClosureFnOnce<FnMut: ?Sized>: 'static {
```

Comment:
Realistically, what are the chances of this beaing a breaking change?

Reply:
Generally speaking `#[doc(hidden)]` is not public API. But I'm also concerned about weird inference issues.
That said, https://github.com/wasm-bindgen/wasm-bindgen/pull/4584#discussion_r2279674296, very much looks like a breaking change to me.